app/models/branch.rb

```diff
@@ -1,8 +1,17 @@
 class Branch < ActiveRecord::Base
+  include Rails.application.routes.url_helpers
   extend FriendlyId
   belongs_to :project
   friendly_id :name, use: :scoped, scope: :project
 
   has_many :metrics, dependent: :destroy
   validates :name, uniqueness: { scope: :project_id }
+
+  def target_url
```

Comment:
Hmmmm I don't like this being here. The biggest smell is that we are including the route helpers here to make it work.